### PR TITLE
Reorganize code in SqlCompiler.Helpers.cs

### PR DESCRIPTION
**Pull request by AlexUstinov**
This PR improves code formatting. Also it removes code duplication and it replaces `if else` chains with `switch` statements for better readability.

### CODE REVIEW

File: Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs [hunk 1]
```diff
@@ -150,187 +157,212 @@ protected SqlExpression GetBooleanColumnExpression(SqlExpression originalExpress
...
+        default:
+          return (SqlUserColumn) column;
+      }
     }
 
     private static bool ShouldUseQueryReference(CompilableProvider origin, SqlProvider compiledSource)
     {
       var sourceSelect = compiledSource.Request.Statement;
-      if (sourceSelect.From==null)
+      if (sourceSelect.From==null) {
```
Comment (alex-kulakov) [inline Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs:217]: Absent spaces

File: Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs [hunk 2]
```diff
@@ -41,20 +40,23 @@ partial class SqlCompiler
       var sqlSources = sources.OfType<SqlProvider>();
 
       var parameterBindings = sqlSources.SelectMany(p => p.Request.ParameterBindings);
-      if (extraBindings!=null)
+      if (extraBindings!=null) {
```
Comment (alex-kulakov) [inline Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs:43]: Absent spaces

File: Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs (hunk 1 above)
Comment (AlexUstinov) [reply Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs]: Fixed.

File: Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs (hunk 2 above)
Comment (AlexUstinov) [reply Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs]: Fixed.

**Pull request merged**